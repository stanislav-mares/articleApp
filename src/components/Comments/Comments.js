import React from 'react';
import PropTypes from 'prop-types';

import dateSort from 'SharedFunc/SharedFunc';

const Comments = (props) => {

  //sorting comments
  let sortedComments = dateSort(props.comments);

  //creating list of comments
  const comments = sortedComments.map((comment, i) => {
    return(
      <li key={i}>
        <p>{comment.text}</p>
        <h4>{`${comment.author}, ${new Date(comment.date).toLocaleString()}`}</h4>
      </li>
    );
  });

  return(
    <div>
      <h3>Comments:</h3>
      <ul>
        {comments}
      </ul>
    </div>
  );
}

Comments.propTypes = {
  comments: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      author: PropTypes.string,
      text: PropTypes.string,
      date: PropTypes.string
    })
  ),
};

export default Comments;
