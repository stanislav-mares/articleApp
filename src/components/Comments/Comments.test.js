import React from 'react';
import enzymeConfig from 'enzymeConfig/enzymeConfig';
import { shallow } from 'enzyme';

import Comments from 'components/Comments/Comments';

describe('<Comments />', () => {

  const comments = [
    { id: 1, author: 'Germán Toro', text: 'Great!', date: '2017-09-11T12:00:39.430Z' },
    { id: 2, author: 'Elena Builes', text: 'Awesome!', date: '2017-09-11T12:30:39.430Z' },
    { id: 3, author: 'Germán Toro', text: 'Splendid!', date: '2017-09-11T14:30:05.919Z' },
    { id: 4, author: 'Elena Builes', text: 'Excelent!', date: '2017-09-11T11:30:05.919Z' },
  ];

  it('Testing if no comments are rendered if no comments are passed', () => {
    const component = shallow(<Comments comments={[]}/>);

    expect(component.find('li').length).toBe(0);
  });

  it('Testing if correct number of comments is rendered when some comments are passed', () => {
    const component = shallow(<Comments comments={comments}/>);

    expect(component.find('li').length).toBe(comments.length);
  });

  it('Testing structure of the Comments component', () => {
    const component = shallow(<Comments comments={[]}/>);

    expect(component).toMatchSnapshot();
  });
});
