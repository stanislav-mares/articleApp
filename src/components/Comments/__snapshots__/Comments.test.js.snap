// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> Testing structure of the Comments component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comments
    comments={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Comments:
        </h3>,
        <ul />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Comments:",
        },
        "ref": null,
        "rendered": "Comments:",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Comments:
          </h3>,
          <ul />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Comments:",
          },
          "ref": null,
          "rendered": "Comments:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
